src/context-service/index.ts

```diff
@@ -44,12 +44,11 @@ export class ContextService {
     return await this.#vectorService.save(args);
   }
 
-  // eslint-disable-next-line @typescript-eslint/require-await
   async addMany(args: AddContextPayload[] | string[]) {
-    return args.map((data) => this.add(data));
+    return Promise.all(args.map((data) => this.add(data)));
```

Comment:
don't we have a batch option? Why do we do it 1 by 1

Reply:
presumably it was done to avoid refactoring the [save method](https://github.com/upstash/rag-chat/blob/1bc3f044c3ea4a47292da888ecfdb2c086fda196/src/database.ts#L143) which formats the AddContextPayload and makes the upsert. To batch in addMany, we would have to split the save method into two and call the first half in addMany.

If would prefer to do this another PR if that's okay with you